test/groovy/CloudFoundryDeployTest.groovy

```diff
@@ -244,8 +249,63 @@ class CloudFoundryDeployTest extends BasePiperTest {
 
         assertThat(jscr.shell, hasItem(containsString('cf login -u "test_cf" -p \'********\' -a https://api.cf.eu10.hana.ondemand.com -o "testOrg" -s "testSpace"')))
         assertThat(jscr.shell, hasItem(containsString("cf blue-green-deploy testAppName --delete-old-apps -f 'test.yml'")))
+        assertThat(jscr.shell, hasItem(containsString("cf logout")))
+
+    }
+
+    @Test
+    void testCfNativeBlueGreenKeepOldInstance() {
+
+        jryr.registerYaml('test.yml', "applications: [[]]")
+
+        jsr.step.cloudFoundryDeploy([
+            script: nullScript,
+            juStabUtils: utils,
+            deployTool: 'cf_native',
+            deployType: 'blue-green',
+            keepOldInstance: true,
+            cfOrg: 'testOrg',
+            cfSpace: 'testSpace',
+            cfCredentialsId: 'test_cfCredentialsId',
+            cfAppName: 'testAppName',
+            cfManifest: 'test.yml'
+        ])
+
+        assertThat(jedr.dockerParams, hasEntry('dockerImage', 's4sdk/docker-cf-cli'))
+        assertThat(jedr.dockerParams, hasEntry('dockerWorkspace', '/home/piper'))
+
+        assertThat(jscr.shell, hasItem(containsString('cf login -u "test_cf" -p \'********\' -a https://api.cf.eu10.hana.ondemand.com -o "testOrg" -s "testSpace"')))
+        assertThat(jscr.shell, hasItem(containsString("cf blue-green-deploy testAppName -f 'test.yml'")))
+        assertThat(jscr.shell, hasItem(containsString("cf stop testAppName")))
+        assertThat(jscr.shell, hasItem(containsString("cf logout")))
     }
 
+    @Test
+    void testCfNativeStandardShouldNotStopInstance() {
+
+        jryr.registerYaml('test.yml', "applications: [[]]")
+
+        jsr.step.cloudFoundryDeploy([
+            script: nullScript,
+            juStabUtils: utils,
+            deployTool: 'cf_native',
+            deployType: 'standard',
+            keepOldInstance: true,
```

Comment:
That's not right. If this test is green, what is it testing anyway?

Reply:
I deleted some assertions. Should be clear now.